### DISCUSSION TITLE
Feature request: Embed/host Discussions as a website forum

### DISCUSSION THREAD

**Discussion by Daeraxa** [community/community#56489] [Discussions]
### Select Topic Area

Product Feedback

### Body

This was touched upon in the recent maintainer Discussions AMA and encouraged to post it here;  essentially this would be a feature request to allow projects to "host" GitHub Discussions as a more traditional forum on their own website. Essentially what this post boils down to is that Discussions is great when the community board is focused on the GitHub or project side of things (open source, collaboration, contribution etc.) but not as good when the community board is application and user focused.

## Other products/services

Other commonly used products for this purpose are things like [Discourse](https://www.discourse.org/) (which obviously this very community board used before) and [Flarum](https://flarum.org/). Mostly I'm referring to the hosted options when drawing comparisons to GitHub Discussions here - they are both open source projects you can host yourselves but that isn't really feature comparable unless GitHub decide to make Discussions somehow hostable on your own infrastructure. So the features are really about how having the product hosted on a third party site can still be integrated with your own project.

There is also [Giscus](https://giscus.app/) which is a commenting system powered by Discussions but really serves a different purpose to what I'm discussing here.

One of the big benefits of using those products over GitHub Discussions is that they can retain a closer tie to the actual website of the project.

For example you can:
- Have a custom domain i.e. the board will be hosted on `forums.myProject.org`
- Add project branding via logos, colour schemes
- Add a website navigation bar to integrate within the rest of the site

For example, here is an image of the retired GitHub Atom forum (I used Atom as it was the only one where I could find a screenshot of a Discourse forum that had been migrated to Discussions and being a dead project it won't be harming any existing communities). You can see that it retains the navigation bar that is themed and relates to the rest of the website along with its own branding on the forum section itself:
![image](https://github.com/community/community/assets/58074586/f98f8224-8729-44fc-a43e-cfd4a697e059)

This really makes the forum feel like it is part of your project when you have your own project website, domain, theme etc. The UI is very clear that this page is all about the discussions, other than the navbar there isn't anything else on the page that isn't "forum".

Now we move to GitHub Discussions for the same project :
![image](https://github.com/community/community/assets/58074586/2245a776-4009-403d-9a47-d7c309353b66)
The biggest things that stand out to me here are:
- We are now on a completely different website - all familiarity with the project website navigation has now gone
- Almost no project branding - almost none of the original website colour scheme or iconography has made the transition. The primary branding is all the GitHub stuff at the very top.
- Unnecessary information - Overlaying the discussions is all the project stuff - issues, pull requests.
- Two search boxes, I can definitely see people trying to use the big one at the top of the screen to search the forum and getting very confused.

## Different types of users and their expectations

This is particularly important when taking into account community members who are only really interested in the application itself as a user. There are people who use open source projects but don't really care about FOSS, open-source as a concept or getting involved with development or issues. I have seen complaints that people don't want to log bugs on GitHub because it is "too techy". There is the infamous [LTT video](https://youtu.be/3E8IGy6I9Wo) outright claiming that "GitHub is for developers and not for end users", this is not an isolated case of people claiming that. 

Most of us here are used to the GitHub UI but for a new user who doesn't really "use" GitHub, this is a lot of unrelated stuff that might simply be pushing people away. I also can see that due to the lack of branding presented on the Discussions page that people might well think that the entire project somehow belongs to GitHub or is otherwise affiliated or endorsed.

Granted the "org" Discussions is much better in this sense. It shows the org logo at the top, has a better layout etc. However it still has a bunch of stuff that many forum goers just don't care about (and the org might not even use) - Packages? Projects?

## How could this look?

Now I'm not saying that all GitHub references and integrations should be hidden - after all one of the main benefits to using Discussions is that you don't have to worry about user administration in the same way. Accounts are all handled by GitHub so unlike Discourse etc. we don't have to worry about storing any user data, data breaches, email accounts, account removal requests etc. Slap some GitHub logos in there next to the sign up/in section. Throw in a `powered by [GitHub Discussions](https://github.com/features/discussions)` or something.

So as a horrible mockup, what I would personally love to see is to make the following possible (again using Atom as the example here and it should be blatantly obvious that I'm not a graphics designer as I just fabricobble all the various website elements together...).

Such a page would be hosted at something like `discussions.atom.io` or `atom.io/discussions`.

![image](https://github.com/community/community/assets/58074586/8d11b4c2-aad9-4296-b858-25598e25b056)

As for how this could be implemented? Absolutely no idea. Via some kind of GitHub pages like service? Part of GitHub pages? (for example `myProject.github.io/discussions`).

Obviously one should still be able to get here via the "traditional" route and directly navigate from the repo or org to the Discussions page using the existing tab *without* being taken "away" from GitHub.

This went on for far longer than I thought so if you read this far, thanks for putting up with my ramblings. Interested to see any comments you might have on this as I've certainly overlooked an awful lot of stuff.

**Comment by Jake7208** [comment]
This sounds interesting but I'm a little confused, is it like a way to use discussions in your own application only showing the discussions for that webpage/website so you keep the discussions feature but only the discussions for your webpage? 

And if so do you think it would be possible to use it kind of like a github api for example how you do validation through github with the github API

**Reply by Daeraxa** [reply]
Basically yes.
In my naive mind it would work a bit like GitHub Pages but with this and only this page included by default. Allow one to configure a custom domain name and the page can be served in such a way that it doesn't feature all the GitHub navigation stuff and allows one to configure their own branding/navbar etc. on the page.

Not sure about the latter, I wouldn't have thought so.

**Reply by Jake7208** [reply]
That sounds like a cool api to me man!

**Comment by osseonews** [comment]
this is a great idea. Any work on this? I love github discussions and would like to use instead of something like discourse.
